includes/entry.inc.php

```diff
@@ -134,22 +134,17 @@
 	
 	// Select data for thread-tree
 	$thread = $entrydata['tid'];
-	// Override '$display_spam_query_and' variable, which was set in main.inc.php, to display current message in tree
-	if ($entrydata['spam'] == 1 && isset($id))
-		$display_spam_query_and .= " OR `ft`.`id` = " . intval($id);
-	$result = mysqli_query($connid, "SELECT id, pid, tid, ft.user_id, UNIX_TIMESTAMP(ft.time) AS time, UNIX_TIMESTAMP(ft.time + INTERVAL " . $time_difference . " MINUTE) AS disp_time,
-                        UNIX_TIMESTAMP(last_reply) AS last_reply, name, user_name, user_type, subject, category, marked, text, rst.user_id AS req_user,
-						" . $db_settings['akismet_rating_table'] . ".spam AS akismet_spam,
-						" . $db_settings['b8_rating_table'] . ".spam AS b8_spam,
-						" . $db_settings['akismet_rating_table'] . ".spam_check_status AS akismet_checked,
-						" . $db_settings['b8_rating_table'] . ".training_type AS b8_checked
-						FROM " . $db_settings['forum_table'] . " AS ft
-                        LEFT JOIN " . $db_settings['userdata_table'] . " ON " . $db_settings['userdata_table'] . ".user_id = ft.user_id
-                        LEFT JOIN " . $db_settings['read_status_table'] . " AS rst ON rst.posting_id = ft.id AND rst.user_id = " . intval($tmp_user_id) . "
-						LEFT JOIN " . $db_settings['akismet_rating_table'] . " ON " . $db_settings['akismet_rating_table'] . ".`eid` = `ft`.`id` 
-						LEFT JOIN " . $db_settings['b8_rating_table'] . " ON " . $db_settings['b8_rating_table'] . ".`eid` = `ft`.`id` 
-                        WHERE tid = " . $thread . $display_spam_query_and . " ORDER BY time ASC");
-	
+	// Override spam variable, which was set in main.inc.php, to display current message in tree
+	if ($entrydata['spam'] == 1 && isset($id)) {
+		$spam_sql_and .= " OR `ft`.`id` = " . intval($id);
+	}
+	$entry_sql = 
+		"SELECT ft.id, ft.pid, ft.tid, ft.user_id, UNIX_TIMESTAMP(ft.time) AS time, UNIX_TIMESTAMP(ft.time + INTERVAL " . $time_difference . " MINUTE) AS disp_time, UNIX_TIMESTAMP(last_reply) AS last_reply, name, user_name, user_type, subject, category, marked, text, rst.user_id AS req_user, " . $db_settings['akismet_rating_table'] . ".spam AS akismet_spam, " . $db_settings['b8_rating_table'] . ".spam AS b8_spam, " . $db_settings['akismet_rating_table'] . ".spam_check_status AS akismet_checked, " . $db_settings['b8_rating_table'] . ".training_type AS b8_checked 
+		FROM " . $db_settings['forum_table'] . " AS ft LEFT JOIN " . $db_settings['userdata_table'] . " ON " . $db_settings['userdata_table'] . ".user_id = ft.user_id LEFT JOIN " . $db_settings['read_status_table'] . " AS rst ON rst.posting_id = ft.id AND rst.user_id = " . intval($tmp_user_id) . " LEFT JOIN " . $db_settings['akismet_rating_table'] . " ON " . $db_settings['akismet_rating_table'] . ".eid = ft.id LEFT JOIN " . $db_settings['b8_rating_table'] . " ON " . $db_settings['b8_rating_table'] . ".eid = ft.id 
```

Comment:
One question: is it really necessary to join the akismet-rating and b8-rating tables individually and to join them again in the next line as a unionised subquery with the same criteria?

Reply:
Ahh, very good question. My simple answer is yes, and even though it's a bit weird looking, it is extremely performant. I'll explain my approach and maybe we can find cleaner solution.

1) The queries need to be performant.
2) Through my testing, creating the subquery which retrieves only the spam = 1 records is always far more performant than any method which retrieves spam = 0 records.
3) I wanted all of the queries I modified to be as consistent in their logic as possible, as long as #1 above is satisfied.
4) The basic challenge is that these queries for entry.inc.php and thread.inc.php require additional fields that the other modified queries don't (akismet_spam, b8_spam, akismet_checked, b8_checked). Retrieving these fields within the inner query UNION could create multiple records for a single entry, as it is quite possible the spam values are different in the two tables. Plus, we need to individually retrieve the checked status for both.
5) As an alternative, I tried to retrieve all required fields within the subquery using a SELECT on mlf2_entries WITH LEFT JOINs on akismet and b8 and WHERE akismet spam = 1 OR b8 spam = 1.  This resulted in another slow performing query, so I abandoned that approach.  
6) Assuming that the subquery as currently modified always retrieves a limited number of results, adding the JOIN to akismet and b8 in the outer query to get the additional fields only retrieves a few records, and they are retrieved via index. This makes it an efficient approach.

I'm always open to looking at a different approach, but this was the result that was performant across all queries and retrieved the same results as 20240308.1.

Note: 
I'll also take a look at your other comments and make the appropriate changes.  I completely agree about handling these queries in a uniform way, and consistent with what is already in production.  In fact when making my changes I made a point to clean up some of the sql code from 20240308.1 to do just that.

Joe

Reply:
Thank you very much for your detailed explanation.